cmd/build/v2/build.go

```diff
@@ -253,6 +260,39 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 	return options.Manifest.ExpandEnvVars()
 }
 
+func (ob *OktetoBuilder) calculateBuildContextHash(buildInfo *model.BuildInfo) string {
```

Comment:
can't we re-use the `getBuildHashFromCommit` but with `treeHash` instead of `repoCommit` ?

Reply:
as line 202

Reply:
I thought about it but some fields in the generated hash has values related to the hash like `commit` vs `tree_hash` that's why I split it. I could inject this value to use only one func.

Reply:
cebae20a007361f6ea2f43a51e8d273c7b0e78eb to reuse same func (sorry for not isolate the change)

Reply:
but... we need to use a commit, instead of the one on the root project, the one of the build context

Reply:
we are using it. https://github.com/okteto/okteto/pull/4023/files#diff-291c33d5347afee4c757a4c6efd49c310488a2fc7855fc15187108eda8202293R90-R97 . Is that what you mean?

Reply:
build hash key is "commit" to build the hash using the repo commit [(like the current one)](https://github.com/okteto/okteto/pull/4023/files#diff-b219a27123f02bc2d87bec867d81a32e91c18ce5edd0dff86aabec32c7647241R461-R463) and "tree_hash" for build hash made by the tree_hash (build context)

Reply:
input is only `tree_hash` for build context hashes